### PR TITLE
Feature/inheart 99

### PR DESCRIPTION
**Pull request by pmusialik-intive**
Added docker with postgres, created CRUD for auctions.

Docker needed. Steps included in readme.

After installing docker and compose, run this command in your terminal:
docker compose --env-file ./.env.dev up

If you want, you can create .env file with the same content as .env.dev and you can run just 
docker compose up

image

Insomnia requests:

POST http://localhost:3333/api/auctions
{
	"title": "Auction title",
	"category": "Category"
}

image

GET http://localhost:3333/api/auctions
image

GET http://localhost:3333/api/auctions/1
image

DELETE http://localhost:3333/api/auctions/1
image

(another) POST http://localhost:3333/api/auctions 
image

PATCH http://localhost:3333/api/auctions/2
image

(final) GET http://localhost:3333/api/auctions
image

### CODE REVIEW

**Review by KonradKarlinski — APPROVED** [review]
For me is look okey.

File: apps/backend/src/app/auctions/auctions.controller.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  }
+
+  @Get()
+  findAll() {
+    return this.auctionsService.findAll();
+  }
+
+  @Get(':id')
+  findOne(@Param('id') id: string) {
+    return this.auctionsService.findOne(+id);
```
Comment (cierzniak) [inline apps/backend/src/app/auctions/auctions.controller.ts:22]: Do we use integer values? Or numeric? This +ID converts to number.

If we use numeric, you can add pipeline to verify it's proper number (positive and numeric).

File: apps/backend/src/app/auctions/auctions.service.ts [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+  findAll() {
+    return this.auctionsRepository.find();
+  }
+
+  async findOne(id: number) {
+    const auction = await this.auctionsRepository.findOneBy({ id });
+
+    if (!auction) {
+      throw new NotFoundException(`Auction #${id} not found`);
```
Comment (cierzniak) [inline apps/backend/src/app/auctions/auctions.service.ts:29]: I'm not sure we should throw http exceptions from service. This is how things happen in nestjs? 😎

**Review by cierzniak — APPROVED** [review]
For.me it's OK 👌 good job 👍 two minor issues raised

File: apps/backend/src/app/auctions/auctions.controller.ts (hunk 1 above)
Comment (pmusialik-intive) [reply apps/backend/src/app/auctions/auctions.controller.ts:22]: I wanted to add pipelines in the next PR (or PRs) 🤓

File: apps/backend/src/app/auctions/auctions.service.ts (hunk 2 above)
Comment (pmusialik-intive) [reply apps/backend/src/app/auctions/auctions.service.ts:29]: Alright, took it from the tutorial, will think about it 😄

**Pull request merged**